presto-main/src/main/java/com/facebook/presto/metadata/SqlAggregationFunction.java

```diff
@@ -91,39 +102,27 @@ public boolean isDeterministic()
 
     public abstract InternalAggregationFunction specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry);
 
-    public static class SimpleSqlAggregationFunction
+    public static class AnnotationBasedSqlAggregationFunction
             extends SqlAggregationFunction
     {
-        private final InternalAggregationFunction function;
-        private final String description;
+        private final BindableAggregationFunction aggregationSignature;
```

Comment:
`AnnotationBased` doesn't seem like the right name. I'd name this `SqlAggregationFunctionAdapter` or something like that